dart/dynamics/BallJoint.cpp

```diff
@@ -71,6 +71,16 @@ const std::string& BallJoint::getStaticType()
 }
 
 //==============================================================================
+void BallJoint::setTransformFromChildBodyNode(const Eigen::Isometry3d& T)
+{
+  Joint::setTransformFromChildBodyNode(T);
+
+  // We update Jacobian here since the Jacobian of BallJoint is a function of
+  // "T" but not of joint positions.
+  mJacobian = math::getAdTMatrix(T).leftCols<3>();
```

Comment:
Same thing here: I think this should be moved to the ``updateLocalJacobian()`` function and wrapped in the ``_mandatory`` flag.